File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(n: int) -> int:
    """Sum numbers [1..n] and return the square."""
    return (n * (n + 1) // 2) ** 2


def sum_of_squares(n: int) -> int:
    """Square numbers [1..n] and return the sum."""
    return n * (n + 1) * (2 * n + 1) // 6


def difference_of_squares(n: int) -> int:
    """Calculate the difference between the sum of squares and squared sum of [1..n]."""
    return (3 * n ** 2 + 2 * n) * (n ** 2 - 1) // 12

File: kotlin/src/main/kotlin/difference_of_squares/DifferenceOfSquares.kt
package difference_of_squares

internal class Squares(val n: Long) {

    fun sumOfSquares() = (2 * n + 1) * n * (n + 1) / 6

    fun squareOfSum() = (n * (n + 1) / 2).let { it * it }

    fun difference() = squareOfSum() - sumOfSquares()
}
